src/compiler/transformers/declarations/localInferenceResolver.ts

```diff
@@ -340,162 +344,313 @@ export function createLocalInferenceResolver({
+                    member.typeParameters,
+                    member.parameters,
+                    type,
+                );
+                if (member.modifiers?.some(modifier => modifier.kind === SyntaxKind.StaticKeyword)) {
+                    staticMembers.push(methodSignature);
+                }
+                else {
+                    nonStaticMembers.push(methodSignature);
+                }
+            }
+            else if (isGetAccessorDeclaration(member) || isSetAccessorDeclaration(member)) {
+                if (!hasGetSetAccessor) {
+                    hasGetSetAccessor = true;
+                    let type;
+                    if (isGetAccessorDeclaration(member)) {
+                        type = localInferenceFromInitializer(member, member.type);
+                    }
+                    else {
+                        type = localInferenceFromInitializer(member.parameters[0], member.parameters[0].type);
+                    }
+                    if (!type) {
+                        invalid = true;
+                    }
+                    nonStaticMembers.push(
+                        factory.createPropertySignature(
+                            [],
+                            member.name,
+                            /*questionToken*/ undefined,
+                            type,
+                        ),
+                    );
+                }
+            }
+            else if (isIndexSignatureDeclaration(member)) {
+                nonStaticMembers.push(member);
+            }
+            else if (isPropertyDeclaration(member)) {
+                const name = isPrivateIdentifier(member.name) ?
+                    // imitating the behavior from utilities.ts : getSymbolNameForPrivateIdentifier, but as we don't have
+                    // a Symbol & SymbolId in hand, we use NodeId of the declaration instead as an approximiation and to provide uniqueness.
+                    // TODO: This seems to have a high collision possibilitiy than the vanilla implementation as we have much less
+                    // ids for nodes in DTE.
+                    factory.createStringLiteral(`__#${node.parent.id}@${member.name.escapedText}`) :
+                    member.name;
+                const type = localInferenceFromInitializer(member, member.type);
+                if (!type) {
+                    invalid = true;
+                }
+                const propertySignature = factory.createPropertySignature(
+                    keepReadonlyKeyword(member.modifiers),
+                    name,
+                    member.questionToken,
+                    type,
+                );
+                if (member.modifiers?.some(modifier => modifier.kind === SyntaxKind.StaticKeyword)) {
+                    staticMembers.push(propertySignature);
+                }
+                else {
+                    nonStaticMembers.push(propertySignature);
+                }

```

Comment:
Similar to Property assignment from object literals, again maybe we can reuse ?

Reply:
Done.